electrum-client/ElectrumClient.ts

```diff
@@ -605,4 +624,41 @@ export class ElectrumClient {
             tx.inputs.some(input => input.address && addresses.has(input.address))
             || tx.outputs.some(output => output.address && addresses.has(output.address)));
     }
+
+    private async callAgents<R>(call: AgentCall<R>, onAgentError?: AgentErrorCallback, stopAfterFirstResult?: true): Promise<R>;
+    private async callAgents<R>(call: AgentCall<R>, onAgentError: AgentErrorCallback | undefined, stopAfterFirstResult: false): Promise<R[]>;
+    private async callAgents<R>(call: AgentCall<R>, onAgentError?: AgentErrorCallback, stopAfterFirstResult: boolean = true): Promise<R | R[]> {
```

Comment:
I don't like that there is an optional third parameter that can be overlooked. I would prefer two separate functions, one for each use case: `callFirstAgent` and `callAllAgents`. That way the behavior is clear from the naming.